Mathlib/LinearAlgebra/Eigenspace/Matrix.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+
+/-!
+
+# Eigenvalues, Eigenvectors and Spectrum for Matrices
+
+This file collects results about eigenvectors, eigenvalues and spectrum specific to matrices
+over a field.
+
+## Tags
+eigenspace, eigenvector, eigenvalue, eigen, spectrum, matrices, matrix
+
+-/
+
+section SpectrumDiagonal
+
+variable {R : Type*} [Field R] {n : Type*} [DecidableEq n] [Fintype n] (M : Type*)
```

Comment:
I don't think we really want `M` explicit here. Notice also that we generally group `Type` arguments at the beginning.
```suggestion
variable {R n M : Type*} [Field R] [DecidableEq n] [Fintype n]
```

Reply:
Neat. I didn't know that. Thanks!